src/app/core/rpc/rpc-state.class.ts

```diff
@@ -38,20 +39,40 @@ export class RPCStateClass {
 
   private blockLoop() {
     if (this.rpc.state.get('blocks') === 0) {
-      setTimeout(this.blockLoop, 1000);
+      setTimeout(this.blockLoop.bind(this), 1000);
     }
     this.rpc.stateCall('getblockchaininfo');
   }
 
   private walletLockedState() {
     this.rpc.state.observe('encryptionstatus')
-      .subscribe(status => this.rpc.state
-        .set('locked', ['Locked', 'Unlocked, staking only'].includes(status)));
+    .subscribe(status => this.rpc.state
+      .set('locked', ['Locked', 'Unlocked, staking only'].includes(status)));
   }
 
-  private initWalletState() {
+  private coldStakeLoop() {
+    if (this.rpc.state.get('locked') === false) {
+      // only available if unlocked
+      this.rpc.call('walletsettings', ['changeaddress'])
+      .subscribe(
+        response => {
+        // check if account is active
+        if (response.changeaddress === 'default') {
```

Comment:
`response => this.rpc.state.set('coldstaking', response.changeaddress === 'default' ? false : !!response.changeaddress.coldstakingaddress)`

Reply:
Will do